### PR TITLE
Fix VerifyStateIndependent

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 1]
```diff
@@ -400,6 +400,7 @@ public virtual VerifyResult VerifyStateIndependent(ProtocolSettings settings)
             {
```
Comment (shargon) [inline src/neo/Network/P2P/Payloads/Transaction.cs:400]: do it here?

Comment (erikzhang) [reply src/neo/Network/P2P/Payloads/Transaction.cs:400]: The script cam be empty.

**Comment by roman-khimov** [comment]
This should solve the problem, but I think it's worth adding a test for this.

**Review by shargon — DISMISSED** [review]

**Review by shargon — DISMISSED** [review]

**Review by shargon — APPROVED** [review]

**Review by roman-khimov — APPROVED** [review]

File: src/neo/Network/P2P/Payloads/Transaction.cs [hunk 2]
```diff
@@ -400,6 +400,7 @@ public virtual VerifyResult VerifyStateIndependent(ProtocolSettings settings)
...
                 if (witnesses[i].VerificationScript.IsSignatureContract())
```
Comment (Qiao-Jin) [inline src/neo/Network/P2P/Payloads/Transaction.cs:401]: Do we need to judge whether witnesses is null and the length of it == hashes?

**Review by fyrchik — APPROVED** [review]

**Pull request merged**